host/libcontainer_backend.go

```diff
@@ -1386,3 +1555,45 @@ func createCGroupPartition(name string, cpuShares int64) error {
 	}
 	return nil
 }
+
+type tmpfsCache struct {
+	cache map[int64]string
+	mtx   sync.Mutex
+}
+
+func (t *tmpfsCache) Create(size int64) (string, error) {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	if path, ok := t.cache[size]; ok {
+		return path, nil
+	}
+
+	f, err := ioutil.TempFile("", "flynn-ext2-")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	shutdown.BeforeExit(func() { os.Remove(f.Name()) })
+
+	if err := f.Truncate(size); err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("mkfs.ext2", "-F", "-L", "rootfs", "-m", "0", f.Name())
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return "", fmt.Errorf("error creating ext2 filesystem: %s: %s", err, out)
+	}
+
+	t.cache[size] = f.Name()
```

Comment:
I think we should only cache  the default size, to prevent unbounded cache growth. In the future we can evaluate common sizes and add them or do some heuristics.